src/header.rs

```diff
@@ -64,6 +65,12 @@ pub struct Header {
     #[serde(skip_serializing_if = "Option::is_none")]
     #[serde(rename = "x5t#S256")]
     pub x5t_s256: Option<String>,
+
+    /// Any additional non-standard headers not defined in [RFC7515#4.1](https://datatracker.ietf.org/doc/html/rfc7515#section-4.1).
+    /// Once serialized, all keys will be converted to fields at the root level of the header payload
```

Comment:
Do we want to expose that for serialization? Most of the cases I've seen are people needing to read values in the header I believe

Reply:
I'm not sure what you mean. Can you expand?

Reply:
In my mind, the extras values were only usable for deserialization, eg you get a jwt with some random non compliant things in there but then I remember things like https://github.com/Keats/jsonwebtoken/issues/368#issuecomment-2305990342

Reply:
Im not sure what Vincent means either. `flatten` works both ways: ser/deser. 🤔

Reply:
I thought the extra fields were mostly used for JWTs coming from services and you would not need to actually expose serializing them at all to the user. In practice it seems some provider like Apple do expect you to add custom headers so maybe you do need to handle Serialize after all so no changes to that PR are needed I think